### PR TITLE
Handle delete of empty list of keys

### PR DESCRIPTION
**Pull request by haroldsultan**
Handle deletion of an empty list of keys

### CODE REVIEW

**Comment by haroldsultan** [comment]
Please advise as to how to test.  Thanks

File: lib/redis.rb [hunk 1]
```diff
@@ -555,6 +555,10 @@ def migrate(key, options)
   # @param [String, Array<String>] keys
   # @return [Integer] number of keys that were deleted
   def del(*keys)
+    if keys.is_a?(Array) && keys.empty?
+      return 0
+    end
```
Comment (byroot) [inline lib/redis.rb:560]:
```suggestion
    return 0 if keys.empty?
```

`keys` can't be anything else than an array.

**Comment by byroot** [comment]
> Please advise as to how to test. Thanks

https://github.com/redis/redis-rb/blob/6542934f01b9c390ee450bd372209a04bc3a239b/test/commands_on_value_types_test.rb#L10-L24

File: test/commands_on_value_types_test.rb [hunk 2]
```diff
@@ -30,6 +32,8 @@ def test_del_with_array_argument
 
     assert_equal ["bar", "baz", "foo"], r.keys("*").sort
 
+    assert_equal 0, r.del([])
```
Comment (byroot) [inline test/commands_on_value_types_test.rb:35]: This isn't handled by your code, you need to add a `keys.flatten!` in `del`.

Comment (byroot) [reply test/commands_on_value_types_test.rb:35]: Or even `flatten!(1)`

**Comment by haroldsultan** [comment]
Thanks @byroot for the review and feedback!

**Pull request merged**